sessiontxn/isolation/base.go

```diff
@@ -197,8 +198,16 @@ func (p *baseTxnContextProvider) isTidbSnapshotEnabled() bool {
 	return p.sctx.GetSessionVars().SnapshotTS != 0
 }
 
+// isBeginStmtWithStaleRead indicate whether the current statement is `BeginStmt` with stale read
+// Because stale read will use `staleread.StalenessTxnContextProvider` for query, so if `staleread.IsStmtStaleness()`
+// returns true in other providers, it means the current statement is `BeginStmt` with stale read
+func (p *baseTxnContextProvider) isBeginStmtWithStaleRead() bool {
+	return staleread.IsStmtStaleness(p.sctx)
```

Comment:
Where's the logic checking `isBeiginStmt`, the function name is a bit confusing.